pkg/accesscontrol/policy_rule_index.go

```diff
@@ -181,8 +181,9 @@ func (p *policyRuleIndex) getRoleBindings(subjectName string) []*rbacv1.RoleBind
 
 // getRoleRefs gathers rules from roles granted to a given subject through RoleBindings and ClusterRoleBindings
 func (p *policyRuleIndex) getRoleRefs(subjectName string) subjectGrants {
-	var clusterRoleBindings []roleRef
-	for _, crb := range p.getClusterRoleBindings(subjectName) {
+	crbs := p.getClusterRoleBindings(subjectName)
+	clusterRoleBindings := make([]roleRef, 0, len(crbs))
+	for _, crb := range crbs {
```

Comment:
This is a small optimization, as I also observed `growSlice` taking a fraction of the CPU.

Reply:
If you're optimising for performance, this can use assignment rather than appending...

A very quick benchmark suggests that it should be around 5% faster 😄